crates/libs/result/src/error.rs

```diff
@@ -13,35 +14,54 @@ impl Error {
     pub const fn empty() -> Self {
         Self {
             code: HRESULT(0),
+            #[cfg(windows)]
             info: None,
         }
     }
 
     /// Creates a new error object, capturing the stack and other information about the
     /// point of failure.
     pub fn new<T: AsRef<str>>(code: HRESULT, message: T) -> Self {
-        let message: Vec<_> = message.as_ref().encode_utf16().collect();
-
-        if message.is_empty() {
-            Self::from_hresult(code)
-        } else {
-            unsafe {
-                RoOriginateErrorW(code.0, message.len() as u32, message.as_ptr());
+        #[cfg(windows)]
+        {
+            let message: Vec<_> = message.as_ref().encode_utf16().collect();
+            if message.is_empty() {
+                Self::from_hresult(code)
+            } else {
+                unsafe {
+                    RoOriginateErrorW(code.0, message.len() as u32, message.as_ptr());
+                }
+                code.into()
             }
-            code.into()
+        }
+        #[cfg(not(windows))]
+        {
+            let _ = message;
+            Self::from_hresult(code)
         }
     }
 
     /// Creates a new error object with an error code, but without additional error information.
     pub fn from_hresult(code: HRESULT) -> Self {
-        Self { code, info: None }
+        Self {
+            code,
+            #[cfg(windows)]
+            info: None,
+        }
     }
 
     /// Creates a new `Error` from the Win32 error code returned by `GetLastError()`.
     pub fn from_win32() -> Self {
```

Comment:
Shouldn't this whole method be conditional, same as the `as_ptr` method below?

Reply:
I tried that, but there are a lot of calls to this in other crates.  I think ideally this becomes `#[cfg(windows)]` in the future, and possibly gets renamed to `fn last()` or something.  `from_xxx` sounds like a conversion, but this reads from ambient thread state.

Reply:
Sounds good, we can tweak this separately as needed.